meteor/client/lib/EditAttribute.tsx

```diff
@@ -913,6 +919,121 @@ const EditAttributeJson = wrapEditAttribute(
 		}
 	}
 )
+const EditAttributeArray = wrapEditAttribute(
+	class EditAttributeArray extends EditAttributeBase {
+		constructor(props) {
+			super(props)
+
+			this.handleChange = this.handleChange.bind(this)
+			this.handleBlur = this.handleBlur.bind(this)
+			this.handleEscape = this.handleEscape.bind(this)
+		}
+		isArray(strOrg: string): { parsed: any[] } | false {
+			if (!(strOrg + '').trim().length) return { parsed: [] }
+
+			const values: any[] = []
+			const strs = (strOrg + '').split(',')
+
+			for (const str of strs) {
+				// Check that the values in the array are of the right type:
+
+				if (this.props.arrayType === 'boolean') {
+					const parsed = JSON.parse(str)
+					if (typeof parsed !== 'boolean') return false // type check failed
+					values.push(parsed)
+				} else if (this.props.arrayType === 'int') {
+					const parsed = parseInt(str, 10)
+
+					if (Number.isNaN(parsed)) return false // type check failed
+					values.push(parsed)
+				} else if (this.props.arrayType === 'float') {
+					const parsed = parseFloat(str)
+					if (Number.isNaN(parsed)) return false // type check failed
+					values.push(parsed)
+				} else {
+					// if (this.props.arrayType === 'string')
+					let parsed: string
+					try {
+						parsed = JSON.parse(str)
+					} catch (_err) {
+						// ignore error
+						parsed = str + ''
+					}
+					if (typeof parsed !== 'string') return false // type check failed
+					values.push(parsed.trim())
+				}
+			}
+			return { parsed: values }
+		}
+		handleChange(event) {
+			let v = event.target.value
+
+			const arrayObj = this.isArray(v)
+			if (arrayObj) {
+				this.handleEdit(v, arrayObj.parsed)
+				this.setState({
+					valueError: false,
+				})
+			} else {
+				this.handleUpdateButDontSave(v, true)
+			}
+		}
+		handleBlur(event) {
+			let v = event.target.value
+
+			const arrayObj = this.isArray(v)
+			if (arrayObj) {
+				this.handleUpdate(v, arrayObj.parsed)
+				this.setState({
+					valueError: false,
+				})
+			} else {
+				this.handleUpdateButDontSave(v, true)
+				this.setState({
+					valueError: true,
+				})
+			}
+		}
+		handleEscape(event) {
+			let e = event as KeyboardEvent
+			if (e.key === 'Escape') {
+				this.handleDiscard()
+			}
+		}
+		getAttribute() {
+			const value = super.getAttribute()
+			if (Array.isArray(value)) {
+				// const str = JSON.stringify(value, null, 2) // "[...values]"
+				// return str.slice(1, -1) // remove [ and ]
```

Comment:
These comments don't seem to add much and the solution of using `.join()` seems to be much more sensible.